### DISCUSSION TITLE
Using urql/core as NodeJS standalone client

### DISCUSSION THREAD

**Discussion by snaplesec** [urql-graphql/urql#1665] [Q&A] [answered]
I'm trying to execute a mutation in a Node application, but doesn't seem to return anything, and it isn't working, but also throws no errors.
Not sure what is missing with it.

```
import client from './lib/gqlClient'
import { gql } from '@urql/core'

const ADDATTACH = gql`
mutation AddAttachments($attach: [ClaimAttachment]!){
  claimAttachments(attachments: $attach) {
    fnolId
  }
}`

export async function putAttachments(att) {
    const atts = att
    // @ts-ignore
    const res = await client.executeMutation(ADDATTACH, { attach: atts })
    console.log({...res})
    return res
}

**Comment by JoviDeCroock** [comment, marked as answer]
`executeMutation` will return a `stream` which I don't think is what you're looking for, alternatively if you want to use promisified stuff you can use [`client.mutate`](https://github.com/FormidableLabs/urql/blob/main/packages/core/src/client.ts#L386-L391)
 
 ```js
 const res = await (client.mutation(ADDATTACH, { attach: atts }).toPromise())
 ```
 
 EDIT: `client.mutate` -->  `client.mutation`

**Reply by snaplesec** [reply]
Ok, that's what I need then, thank you. I'll give that a shot.
I couldn't find anything in the docs to show how.

Thanks

**Reply by snaplesec** [reply]
That didn't work.
Gave this error.
```
(node:2240) UnhandledPromiseRejectionWarning: TypeError: gqlClient_1.default.mutate is not a function
    at Object.putAttachments (C:\Users\SteveNaples\OWL\insuredportal-upload-server\src\addAttachments.ts:14:31)

**Reply by JoviDeCroock** [reply]
Should be [here](https://formidable.com/open-source/urql/docs/basics/core/#one-off-queries-and-mutations)

EDIT: apologies it's `.mutation`

**Reply by snaplesec** [reply]
Oh ok, let me change it.

**Reply by snaplesec** [reply]
How do I define fetch in node setup of urql?
I got the error about fetch not defined.
```
(node:28648) UnhandledPromiseRejectionWarning: ReferenceError: fetch is not defined
    at executeFetch (C:\Users\SteveNaples\OWL\insuredportal-upload-server\node_modules\@urql\core\src\internal\fetchSource.ts:30:13)

**Reply by JoviDeCroock** [reply]
By defining fetch when you create the client or using a polyfill https://formidable.com/open-source/urql/docs/api/core/#client

**Reply by snaplesec** [reply]
Ahhhh.... got it. 
Thanks
